src/ring_theory/ideal/local_ring.lean

```diff
@@ -0,0 +1,243 @@
+/-
+Copyright (c) 2018 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Chris Hughes, Mario Carneiro
+-/
+import algebra.category.CommRing.basic
+import ring_theory.ideal.basic
+
+/-!
+
+# Local rings
+
+Define local rings as commutative rings having a unique maximal ideal.
+
+## Main definitions
+
+* `local_ring`: A predicate on commutative rings, stating that every element `a` is either a unit
+  or `1 - a` is a unit. This is shown to be equivalent to the condition that there exists a unique
+  maximal ideal.
+* `local_ring.maximal_ideal`: The unique maximal ideal for a local rings. Its carrier set is given
+  by the `nonunits` above.
```

Comment:
"above" no longer makes sense here

Reply:
You're right, I originally thought I should move `nonunits` to this file as well, but then decided against it. Shiould be fixed now.